pyproject.toml

```diff
@@ -174,6 +174,39 @@ doc = [
     "intersphinx_registry>=0.2405.27",
 ]
 dev = ["mne[test,doc]", "rcssmin"]
+doc_related = [
+    "alphaCSC",
+    "autoreject",
+    "bycycle",
+    "conpy",
+    "emd",
+    "fooof",
+    "meggie",
+    "mne-ari",
+    "mne-bids-pipeline",
+    "mne-faster",
+    "mne-features",
+    "mne-icalabel",
+    "mne-lsl",
+    "mne-microstates",
+    "mne-nirs",
+    "mne-rsa",
+    "neurodsp",
+    "neurokit2",
+    "niseq",
+    "nitime",
+    "openneuro-py",
+    "pactools",
+    "plotly",
+    "pycrostates",
+    "pyprep",
+    "pyriemann",
+    "python-picard",
+    "sesameeg",
+    "sleepecg",
+    "tensorpac",
+    "yasa",
+]
```

Comment:
Still OK to move these to `circleci_dependencies.sh`?  I'm already not happy with how complicated our optional dependencies groups are, so I'm reluctant to add another one. One less reason for head-scratching when folks view this file.